<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Document</title>
</head>
<body>
  <script>
    new Promise((resolve, reject) => {
        setTimeout(() => {
          resolve();
        }, 1000);
      }).then(() => {
        console.log("我是第一次回调处理");

        return new Promise((resolve, reject) => {
          //请求代码
          setTimeout(() => {
            //请求成功调用这个函数，失败就要调用reject函数,然后then函数就不调用了就可以用catch来捕获异常了
            resolve("helloWorld");
            //reject('error');  
          }, 1000);
        })
      }).then((data) => {
        //处理代码
        console.log(data);
        console.log("我是第二次回调处理");

        return new Promise((resolve, reject) => {
          setTimeout(() => {
            resolve();
          }, 1000);
        })
      }).then(() => {
        console.log("我是第三次回调处理");
      },() =>{console.log("error");
    });//这个捕获异常的函数还可写在then函数里 
  </script>
</body>
</html>